src/main/java/de/tum/cit/aet/artemis/communication/service/AnswerMessageService.java

```diff
@@ -93,8 +89,9 @@ public AnswerPost createAnswerMessage(Long courseId, AnswerPost answerMessage) {
             throw new BadRequestAlertException("A new answer post cannot already have an ID", METIS_ANSWER_POST_ENTITY_NAME, "idexists");
         }
 
-        Conversation conversation = conversationService.isMemberOrCreateForCourseWideElseThrow(answerMessage.getPost().getConversation().getId(), author, Optional.empty())
-                .orElse(conversationRepository.findByIdElseThrow(answerMessage.getPost().getConversation().getId()));
+        var conversationId = answerMessage.getPost().getConversation().getId();
+        var conversation = conversationService.isMemberOrCreateForCourseWideElseThrow(conversationId, author, Optional.empty())
```

Comment:
Optional: Maybe add a comment to explain why this differentiation is necessary. If I would come back here after a while this would not be immediately clear to me.